src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -76,36 +76,16 @@
 #include "utils/syscache.h"
 #include "utils/typcache.h"
 
-
 /* user configuration */
 int ReadFromSecondaries = USE_SECONDARY_NODES_NEVER;
 
-
-/*
```

Comment:
I removed `ShardCacheEntry` because I couldn't come up with a clean way to keep this consistent in a world where multiple `CitusTableCacheEntry`s might exist for a relation id

`LookupShardIndex` was updated to return both an index & the cache entry that index is for. This does mean we end up having to scan the ShardInterval array everytime

Reply:
> `LookupShardIndex` was updated to return both an index & the cache entry that index is for. This does mean we end up having to scan the ShardInterval array everytime

I am not sure if this will be ok. Some of our users have hundreds of thousands of shards.

Reply:
Created a PR for this https://github.com/citusdata/citus/pull/3820